presubmit_tests/single-instance-on-gcp.sh

```diff
@@ -0,0 +1,143 @@
+# For available Prow-injected environment variables, see:
+# https://docs.prow.k8s.io/docs/jobs/#job-environment-variables
+toolkit_zip_file_name="oracle-toolkit-${BUILD_ID}.zip"
+tfvars_file="./presubmit_tests/single-instance.tfvars"
+instance_name="github-presubmit-si-${BUILD_ID}"
+deployment_name="presubmit-si-${BUILD_ID}"
+location="us-central1"
+project="gcp-oracle-benchmarks"
+gcs_source="${gcs_bucket}/${toolkit_zip_file_name}"
+deployment_id="projects/${project}/locations/${location}/deployments/${deployment_name}"
+
+apk add --no-cache zip jq
+
+echo "Zipping CWD into /tmp/${toolkit_zip_file_name} and uploading to ${gcs_bucket}/..."
+zip -r /tmp/"${toolkit_zip_file_name}" . -x ".git*" -x ".terraform*" -x "terraform*" -x OWNERS > /dev/null
+gcloud storage cp /tmp/"${toolkit_zip_file_name}" "${gcs_bucket}/"
+
+sed -i "s|@deployment_name@|$deployment_name|g" "${tfvars_file}"
+sed -i "s|@gcs_source@|$gcs_source|g" "${tfvars_file}"
+sed -i "s|@instance_name@|$instance_name|g" "${tfvars_file}"
+
+echo "Applying Infra Manager deployment: ${deployment_id}"
+gcloud infra-manager deployments apply "${deployment_id}" \
+  --service-account="projects/${project}/serviceAccounts/infra-manager-deployer@${project}.iam.gserviceaccount.com" \
+  --local-source="./terraform" \
+  --inputs-file="${tfvars_file}" || exit 1
+
+
+cleanup() {
+  if [[ -n "${tail_pid}" ]]; then
+    kill "${tail_pid}"
+  fi
+  echo "Cleaning up: deleting ${gcs_source} GCS object and ${deployment_id} Infra Manager deployment..."
+  if gcloud infra-manager deployments describe "${deployment_id}" >/dev/null 2>&1; then
+    gcloud --quiet infra-manager deployments delete "${deployment_id}" 
+  fi
+  if gcloud storage objects describe "${gcs_source}" >/dev/null 2>&1; then
+    gcloud --quiet storage rm "${gcs_source}"
+  fi
+}
+
+trap cleanup SIGINT SIGTERM EXIT
+
+# Extract the id of the control node resource
+# The format is: projects/<project>/zones/<zone>/instances/control-node-<random-suffix>
+control_node_resource_id="$(gcloud infra-manager resources list \
+--deployment="${deployment_name}" \
+--location="${location}" \
+--revision=r-0 \
+--format=json | jq -r '.[] | select(.terraformInfo.address == "google_compute_instance.control_node") | .terraformInfo.id')"
+
+if [[ -z "${control_node_resource_id}" ]]; then
+  echo "Could not retrieve control node's resource ID."
+  exit 1
+fi
+
+control_node_instance_zone="$(echo "${control_node_resource_id}" | cut -d'/' -f4)"
+if [[ -z "${control_node_instance_zone}" ]]; then
+  echo "Could not extract control node's zone."
+  exit 1
+fi
+
+control_node_instance_name="$(echo "${control_node_resource_id}" | cut -d'/' -f6)"
+if [[ -z "${control_node_instance_name}" ]]; then
+  echo "Could not extract control node's name."
+  exit 1
+fi
+
+# Get the instance ID from the instance name
+control_node_instance_id="$(gcloud compute instances describe "${control_node_instance_name}" \
```

Comment:
`gcloud compute instances describe` can accept ${control_node_resource_id} directly;  no need to extract and re-add the name and zone.